src/write.rs

```diff
@@ -75,6 +77,12 @@ where
     T: Absorb<O>,
 {
     fn drop(&mut self) {
+        // If we use `Self::take` then we cannot drop both r_handle and w_handle so
+        // a different cleanup happens there.
+        if self.taken {
+            return;
```

Comment:
There is a test coverage warning that says this wasn't covered by tests. But it gets called when the take functions returns.